modules/index/src/main/java/org/apache/ignite/internal/index/IndexNodeFinishedRwTransactionsChecker.java

```diff
@@ -139,37 +140,37 @@ public <T> T inUpdateRwTxCountLock(Supplier<T> supplier) {
     }
 
     /**
-     * Handles {@link IsNodeReadyToStartBuildingIndexRequest} of {@link IndexMessageGroup}.
+     * Handles {@link IsNodeFinishedRwTransactionsStartedBeforeRequest} of {@link IndexMessageGroup}.
      *
      * @see NetworkMessageHandler#onReceived(NetworkMessage, String, Long)
      */
     private void onReceiveIndexNetworkMessage(NetworkMessage message, String senderConsistentId, @Nullable Long correlationId) {
         inBusyLock(busyLock, () -> {
-            if (!(message instanceof IsNodeReadyToStartBuildingIndexRequest)) {
+            if (!(message instanceof IsNodeFinishedRwTransactionsStartedBeforeRequest)) {
                 return;
             }
 
             assert correlationId != null : senderConsistentId;
 
-            int indexAppearanceCatalogVersion = ((IsNodeReadyToStartBuildingIndexRequest) message).catalogVersion();
+            int targetCatalogVersion = ((IsNodeFinishedRwTransactionsStartedBeforeRequest) message).targetCatalogVersion();
 
-            boolean ready = isNodeReadyToStartBuildingIndex(indexAppearanceCatalogVersion);
+            boolean finished = isNodeFinishedRwTransactionsStartedBefore(targetCatalogVersion);
 
             messagingService.respond(
                     senderConsistentId,
-                    FACTORY.isNodeReadyToStartBuildingIndexResponse().ready(ready).build(),
+                    FACTORY.isNodeFinishedRwTransactionsStartedBeforeResponse().finished(finished).build(),
                     correlationId
             );
         });
     }
 
     /**
-     * Returns {@code true} iff the catalog version in which the index appeared is active and all read-write transactions up to the
-     * requested catalog version have been completed on the node.
+     * Returns {@code true} iff the requested catalog version is active and all RW transactions up to that version have finished on the
```

Comment:
It seems that 'up to' means that the upper bound is included, but in our case it's not. Please rephrase, like 'and all RW transactions started on versions strictly before that version'...

Reply:
why not